presto-docs/src/main/sphinx/connector/oracle.rst

```diff
@@ -23,47 +22,363 @@ connection properties as appropriate for your setup:
+ - Precision (``p`` ) is greater than 38. 
+ - Scale is negative and the difference between ``p`` and ``s`` is greater than
+   38, unless ``oracle.number.rounding-mode`` is set to a different value than
+   ``UNNECESSARY``. 
+
+If a column meets any of these conditions, it shows an error:
+
+ - An ``unsupported-type.handling-strategy`` equals ``FAIL``
+   (default configuration) any queries on the table fail.
+ - An ``unsupported-type.handling-strategy`` equals ``IGNORE``
+   (default configuration), Presto no longer accesses the column.
+ - An ``unsupported-type.handling-strategy`` equals ``CONVERT_TO_VARCHAR``
+   (default configuration), a column value converts to an unbounded ``VARCHAR``.
+
+If ``s`` is negative, ``NUMBER(p, s)`` maps to ``DECIMAL(p + s, 0)``.
+
+For Oracle ``NUMBER`` (without precision and scale), you can change
+``oracle.number.default-scale=s`` and map the column to ``DECIMAL(38, s)``.
+
+.. _datetime mapping:
+
+Mapping datetime types
+^^^^^^^^^^^^^^^^^^^^^^
+
+Selecting a timestamp with fractional second precision (``p``) greater than 3
+truncates the fractional seconds to three digits instead of rounding it.
+
+Oracle ``DATE`` values may store hours, minutes, and seconds, mapped
+to Presto ``TIMESTAMP``.
+
+.. warning::
+
+  Due to date and time differences in the libraries used by Presto and the
+  Oracle JDBC driver, attempting to insert or select a datetime value earlier
+  than ``1582-10-15`` results in an incorrect date inserted.
+
+.. _character mapping:
+
+Mapping character types
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Presto's ``VARCHAR(n)`` maps to ``VARCHAR2(n CHAR)`` if ``n`` is no greater than
+4000. A larger or unbounded ``VARCHAR`` maps to ``NCLOB``.
+
+Presto's ``CHAR(n)`` maps to ``CHAR(n CHAR)`` if ``n`` is no greater than 2000.
+A larger ``CHAR`` maps to ``NCLOB``.
+
+Using ``CREATE TABLE AS`` to create an ``NCLOB`` column from a ``CHAR`` value
+removes the trailing spaces from the initial values for the column. Inserting
+``CHAR`` values into existing ``NCLOB`` columns keeps the trailing spaces. For
+example::
+
+    presto> CREATE TABLE vals AS SELECT CAST('A' as CHAR(2001)) col;
+    presto> INSERT INTO vals (col) VALUES (CAST('BB' as CHAR(2001)));
+    presto> SELECT LENGTH(col) FROM vals;
+    2001
+    1
+
+Attempting to write a ``CHAR`` that doesn't fit in the column's actual size
+fails. This is also true for the equivalent ``VARCHAR`` types.
+
+.. note:: 
+
+    This behaviors above may change in a future version.
+
+
+Synonyms
+--------
+
+Based on performance reasons, Presto disables support for Oracle ``SYNONYM``. To
+include ``SYNONYM``, add the following: 
+
+.. code-block:: none
+
+    oracle.synonyms.enabled=true
+
+Pushdown
+--------
+
+The connector supports :doc:`pushdown </optimizer/pushdown>` for optimized query
+processing.
 
 Limitations
 -----------
 
-The following SQL statements are not yet supported:
+Presto supports connecting to Oracle Database 12c. The following SQL statements
```

Comment:
12c is not the only version, 11g works, maybe others too

Reply:
Removed reference in new draft.